File: 800/1607A/main.cpp
#include <bits/stdc++.h>
using namespace std;

int main() 
{
    int t;
    cin >> t;

    while (t--) 
    {
        string a, s;
        cin >> a >> s;

        // Map each character to its position in the custom keyboard layout
        unordered_map<char, int> mp;
        for (int i = 0; i < 26; i++)
            mp[a[i]] = i + 1; // positions are 1-based

        int cost = 0;
        int pos = mp[s[0]];

        for (int i = 0; i < (int)s.size(); i++) 
        {
            cost += abs(mp[s[i]] - pos);
            pos = mp[s[i]];
        }

        cout << cost << "\n";
    }

    return 0;
}

File: 800/1607A/main.java
import java.util.*;
public class main
{
    public static void main(String args[])
    {
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();
        
        while(t>0)
        {
            String a = sc.next();
            String s = sc.next();
            
            HashMap<Character, Integer> mp = new HashMap<Character, Integer>();
            
            for(int i=1;i<=26;i++)
            {
                mp.put(a.charAt(i-1), i);
            }
            
            int cost = 0;
            int pos = mp.get(s.charAt(0));
            
            for(int i=0;i<s.length();i++)
            {
                cost += Math.abs(mp.get(s.charAt(i)) - pos);
                pos = mp.get(s.charAt(i));
            }
            
            System.out.println(cost);
            
            t--;
        }
    }
}